### PR TITLE
Added test to validate Menu options in the Weather Shopper app

### PR DESCRIPTION
**Pull request by shivahari**
1. Added a [test](https://github.com/qxf2/qxf2-page-object-model/blob/validate_weathershopper_app_menu_options/tests/test_weather_shopper_app_menu_options.py) to validate Weather Shopper Menu options
2. Added [helper methods](https://github.com/qxf2/qxf2-page-object-model/blob/validate_weathershopper_app_menu_options/core_helpers/mobile_app_helper.py#L405) to accommodate the test
3. Added methods to the [existing PageObject](https://github.com/qxf2/qxf2-page-object-model/blob/validate_weathershopper_app_menu_options/page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py#L83) to perform actions on the page elements
4. Added a [new PageObject ](https://github.com/qxf2/qxf2-page-object-model/blob/validate_weathershopper_app_menu_options/page_objects/examples/weather_shopper_mobile_app/webview_chrome.py) to handle interactions against Chrome Webview

### CODE REVIEW

File: page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py [hunk 1]
```diff
@@ -5,6 +5,127 @@
...
 from .homepage_objects import HomepageObjects
 from .navigation_menu_objects import NavigationMenuObjects
+import conf.locators_conf as locators
+from utils.Wrapit import Wrapit
 
 class WeatherShopperHomePage(Mobile_App_Helper, HomepageObjects, NavigationMenuObjects):
     "Page objects for home page in Weathershopper application."
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def get_temperature(self):
```
Comment (rohandudam) [inline page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py:15]: Method also available as part of Home Page Objects.

File: page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py [hunk 2]
```diff
@@ -5,6 +5,127 @@
...
+        "This method is to get the temperature in the Weather Shopper application."
+        temperature = locators.temperature
+        current_temperature = self.get_text(temperature)
+        self.write(f"Current temperature is {int(current_temperature)}", level='debug')
+        return int(current_temperature)
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def view_moisturizers(self):
```
Comment (rohandudam) [inline page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py:24]: Method also available as part of Navigation Menu Object.

File: page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py [hunk 3]
```diff
@@ -5,6 +5,127 @@
...
+        self.conditional_write(result_flag,
+            positive='Successfully clicked on Moisturizer tab',
+            negative='Failed to click on Moisturizer tab',
+            level='debug')
+        self.switch_page("weathershopper products page")
+        return result_flag
+
+    @Wrapit._exceptionHandler
+    @Wrapit._screenshot
+    def view_sunscreens(self):
```
Comment (rohandudam) [inline page_objects/examples/weather_shopper_mobile_app/weather_shopper_home_page.py:37]: Method also available as part of Navigation Menu Object.

File: core_helpers/mobile_app_helper.py [hunk 4]
```diff
@@ -390,6 +392,73 @@ def scroll_forward(self, distance=150):
...
+            # This is a fail safe, implemented to address corner case where the context might be avialable after a sec or two
+            for attempt in range(1,retries+1):
+                if not context_name in self.driver.contexts:
+                    self.write(f"Intended context {context_name} not present in available contexts {self.driver.contexts}, retry attempt {attempt}", "debug")
+                    time.sleep(attempt)
+            # Switch context
+            self.driver.switch_to.context(context_name)
+            result_flag = True if self.driver.current_context == context_name else False
+        except NoSuchContextException:
+            self.write(f"No {context_name} context present", 'debug')
```
Comment (rohandudam) [inline core_helpers/mobile_app_helper.py:416]: Please set log level to critical for exceptions. (Please do it for only newly added methods. Others, we did it in another PR.)

**Review by rohandudam — CHANGES_REQUESTED** [review]
Good to make commented changes

File: core_helpers/mobile_app_helper.py (hunk 4 above)
Comment (shivahari) [reply core_helpers/mobile_app_helper.py:416]: The duplicate methods look to be because of a poor rebase @rohandudam. I have fixed it now.

**Review by rohandudam — APPROVED** [review]
Please make log level changes and merge it.

**Comment by shivahari** [comment]
@rohandudam I have modified the logging level for eligible methods I have added.

**Pull request merged**